### PR TITLE
Video in background

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
This is PR for 'Keep videos playing after switching tab, minimizing Brave, switching off display'  https://github.com/brave/browser-android-tabs/issues/160.

Now it asks to restart after the settings has changed.

@jamesmudgett , could you please review strings in grd 
https://github.com/brave/browser-android-tabs/compare/master...AlexeyBarabash:VideoInBackground?expand=1#diff-543312251cada1b8ce4f9fb2b8a5cf1d
and icons images.

### CODE REVIEW

File: chrome/android/java/src/org/chromium/chrome/browser/DisableVideoPauseOnBackground.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+public class DisableVideoPauseOnBackground {
+    public static void Execute(Tab tab) {
+        final boolean videoInBackgroundEnabled = PrefServiceBridge.getInstance().playVideoInBackgroundEnabled();
+        if (videoInBackgroundEnabled && NeedToDisable(tab)) {
+          tab.getWebContents().evaluateJavaScript(SCRIPT, null);
+        }
+    }
+
+    private static boolean NeedToDisable(Tab tab) {
+        boolean bRet = tab != null && tab.getUrl().contains("https://m.youtube.com/watch?v");
```
Comment (SergeyZhukovsky) [inline chrome/android/java/src/org/chromium/chrome/browser/DisableVideoPauseOnBackground.java]: what happens if we open youtube in desktop mode? Do we play the content in background still?

File: chrome/android/java/strings/strings.xml [hunk 2]
```diff
@@ -80,5 +80,8 @@ Saved</string>
     <string name="website_settings_category_play_video_in_background_enabled_summary">Enabled</string>
     <string name="website_settings_category_play_video_in_background_disabled_summary">Disabled</string>
     <string name="website_settings_category_play_video_in_background_enabled">Experimental. Enables audio play from video in background when tab is not active or device screen is turned off. If it is not played still, try to switch page to desktop mode</string>
+    <string name="settings_require_relaunch_notice">Your changes will take effect the next time you restart your device</string>
```
Comment (SergeyZhukovsky) [inline chrome/android/java/strings/strings.xml]: I think we need only restart a browser and not the whole device.

File: chrome/android/java/src/org/chromium/chrome/browser/DisableVideoPauseOnBackground.java (hunk 1 above)
Comment (AlexeyBarabash) [reply chrome/android/java/src/org/chromium/chrome/browser/DisableVideoPauseOnBackground.java]: Yes. Desktop youtube does on put video on pause when the tab gets hidden, so we dont need to patch it with javascript.

File: chrome/android/java/strings/strings.xml (hunk 2 above)
Comment (AlexeyBarabash) [reply chrome/android/java/strings/strings.xml]: Agree. This is mistake.

File: chrome/android/java/src/org/chromium/chrome/browser/DisableVideoPauseOnBackground.java (hunk 1 above)
Comment (SergeyZhukovsky) [reply chrome/android/java/src/org/chromium/chrome/browser/DisableVideoPauseOnBackground.java]: ++

**Comment by D0ve** [comment]
Guys, will this future be optional/switchable? Great if it will so!

**Comment by SergeyZhukovsky** [comment]
@D0ve yes it will be an option

**Comment by D0ve** [comment]
Thank you! And the same about realized in new builds 'playing audio in the background',  is it an option?

**Comment by AlexeyBarabash** [comment]
@D0ve 'playing music/audio in the background' is not an option, it is turned on permanently.

**Comment by D0ve** [comment]
@AlexeyBarabash , it's a pity , imo shoud be possible to turn on/off this future either (.

**Comment by AlexeyBarabash** [comment]
@bbondy  could you please review the strings at
https://github.com/brave/browser-android-tabs/pull/464/files#diff-543312251cada1b8ce4f9fb2b8a5cf1dR3417
if you have a chance.

**Comment by SergeyZhukovsky** [comment]
It is good to merge.

**Pull request merged**